backon/src/lib.rs

```diff
@@ -33,14 +33,42 @@
 //!
 //! Retry in BackON requires an implementation for sleeping. BackON will accept a [`Sleeper`] to pause for a specified duration.
 //!
-//! BackON employs the following default sleep implementations:
+//! ## Default `Sleeper`
 //!
-//! - `tokio-sleep`: Utilizes [`TokioSleeper`] within a Tokio context in non-wasm32 environments.
-//! - `gloo-timers-sleep`: Utilizes [`GlooTimersSleep`] to pause in wasm32 environments.
+//! Currently, BackON has 2 built-in `Sleeper` implementations for different environments,
+//! they are gated under their own features, which are enabled by default:
 //!
-//! Users CAN provide a custom implementation if they prefer not to use the default options.
+//! |      `Sleeper`      | feature          | Environment |
+//! |---------------------|------------------|-------------|
+//! | [`TokioSleeper`]    | tokio-sleep      | non-wasm32  |
+//! | [`GlooTimersSleep`] | gloo-timers-sleep|   wasm32    |
 //!
-//! If neither feature is enabled nor a custom implementation is provided, BackON will fallback to an empty sleeper. This will cause a panic in the `debug` profile and do nothing in the `release` profile.
+//! ## Custom `Sleeper`
+//!
+//! If you do not want to use the built-in `Sleeper`, you CAN provide a custom
+//! implementation, here is an example that implements a `Sleeper` with `monoio::time::sleep`:
+//!
+//! ```rust,ignore
```

Comment:
I ignore this example, but we can actually test it if we add Monoio to our `dev-dependeices`👀

Reply:
This comment no longer applies